### ISSUE TITLE
Nicer way to filter jobs by date

### ISSUE THREAD

**Issue by ajavadia** [Qiskit/qiskit-ibmq-provider#442]
`backend.jobs()` currently accepts a raw `db_filter` that can be used for pretty much any type of filtering, but using it is not always easy because one needs insight into how database queries work.
I think some of the common filtering operations should be promoted to python arguments in `backend.jobs()`. This is already done for `status`. The next one I think should be `date`.

Currently if I want jobs between sunday and monday I have to do:
```
import datetime
monday = datetime.datetime(2019, 11, 11)
sunday = datetime.datetime(2019, 11, 10)
date_filter = {'creationDate': {'lt': monday.isoformat(), 'gt': sunday.isoformat()}}
job_list = backend.jobs(db_filter=date_filter)
```

can we instead do:
```
job_list = backend.jobs(datetime_begin=sunday, datetime_end=monday)
```
or something like that?

**Comment by ajavadia** [comment]
BTW since we now support two types of jobs queries, their API should be consistent:
`backend.jobs()`
`provider.backends.jobs()`
except the 2nd one also allows filtering by backend.

**Comment by jyu00** [comment]
To Ali's second point, it might be worth having something that asserts the signature of the two are mostly equal. It'd also apply to the methods of `Result` and `ManagedResults`.

**Comment by luisg5** [comment]
Closed via #452 and #457.
